app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -71,39 +70,48 @@ class VirtualMachineTemplate < ActiveRecord::Base
 
     scope :active, -> { where(state: 'active') }
 
-    scope :on_active_cs, -> do
-      joins(:tenant).
-        where(atmosphere_tenants: { active: true })
+    scope :on_active_tenant, -> do
+      joins(:tenants)
+        .where('atmosphere_tenants.active = ?', true)
     end
 
-    scope :on_cs, ->(t) { where(tenant_id: t) }
+    scope :on_tenant, ->(t) do
+      joins(:tenants)
+        .where('atmosphere_tenants.id = ?', t)
+    end
 
-    scope :on_cs_with_src, ->(t_id, source_id) do
-      joins(:tenant).
-        where(atmosphere_tenants: { tenant_id: t_id },
-              id_at_site: source_id)
+    scope :on_tenant_with_src, ->(t_id, source_id) do
+      joins(:tenants)
+        .where("atmosphere_tenants.tenant_id = '#{t_id}' AND
+        id_at_site = '#{source_id}'")
     end
 
     def uuid
-      "#{tenant.tenant_id}-tmpl-#{id_at_site}"
+      if tenants.blank?
+        "NOTENANT-tmpl-#{id_at_site}"
+      else
+        # Kinda hacky - assumes all vmt.tenants share the same site_id.
+        "#{tenants.first.site_id}-tmpl-#{id_at_site}"
+      end
     end
 
     def self.create_from_vm(virtual_machine, name = virtual_machine.name)
       name_with_timestamp = "#{name}/#{VirtualMachineTemplate.generate_timestamp}"
       tmpl_name = VirtualMachineTemplate.sanitize_tmpl_name(name_with_timestamp)
-      cs = virtual_machine.tenant
+      t = virtual_machine.tenant
 
-      id_at_site = cs.cloud_client
+      id_at_site = t.cloud_client
         .save_template(virtual_machine.id_at_site, tmpl_name)
       logger.info "Created template #{id_at_site}"
 
-      vm_template = cs.virtual_machine_templates
+      vm_template = t.virtual_machine_templates
         .find_or_initialize_by(id_at_site: id_at_site)
 
       vm_template.source_vm = virtual_machine
       vm_template.name = tmpl_name
       vm_template.managed_by_atmosphere = true
       vm_template.state = :saving
+      vm_template.tenants = [t]
```

Comment:
No, this would automatically trigger a `save` operation on `vm_template` and I don't want to do that just yet.